vortex-array/src/compute/compare.rs

```diff
@@ -262,21 +262,13 @@ mod tests {
 
     fn to_int_indices(indices_bits: BoolArray) -> Vec<u64> {
         let buffer = indices_bits.boolean_buffer();
-        let null_buffer = indices_bits
-            .validity()
-            .to_logical(indices_bits.len())
-            .unwrap()
-            .to_null_buffer();
-        let is_valid = |idx: usize| match null_buffer.as_ref() {
-            None => true,
-            Some(buffer) => buffer.is_valid(idx),
-        };
-        let filtered = buffer
+        let mask = indices_bits.validity_mask().unwrap();
+        let is_valid = |idx: usize| mask.value(idx);
```

Comment:
Why the closure?

Reply:
It should be gone. I didn’t simplify this fully